### PR TITLE
Support img tag rendering for text

### PR DESCRIPTION
**Pull request by wx-uf**
1. add img, br, p and span tags to be allowed to render html string
2. add test case to cover the change

### CODE REVIEW

File: src/components/elements/text/text.tsx [hunk 1]
```diff
@@ -71,9 +71,10 @@ export const Text = (props: IGenericElementProps<ITextSchema>) => {
 				);
 			});
 		} else if (typeof children === "object") {
-			return Object.entries(children).map(([id, childSchema], index) => (
-				<Text key={index} id={id} schema={childSchema} />
-			));
+			return Object.entries(children).map(([id, childSchema], index) => {
+				console.log("----a----", id, childSchema);
```
Comment (weili-govtech) [inline src/components/elements/text/text.tsx:75]: remove console.log

File: src/components/elements/text/text.tsx [hunk 2]
```diff
@@ -89,7 +90,17 @@ export const Text = (props: IGenericElementProps<ITextSchema>) => {
...
+					inline
+					sanitizeOptions={{
+						allowedTags: ["img", "br", "p", "span"],
+						allowedAttributes: {
+							img: ["src", "alt", "width", "height"],
+							p: ["class", "data-testid", "id"],
+							span: ["class", "data-testid", "id"],
+						},
+					}}
+				>
```
Comment (weili-govtech) [inline src/components/elements/text/text.tsx:103]: the default allowed tags and attributes are a lot more exhaustive, you can refer to their [docs](https://github.com/apostrophecms/sanitize-html?tab=readme-ov-file#i-like-your-set-but-i-want-to-add-one-more-tag-is-there-a-convenient-way) to see how to extend from their default list

**Review by weili-govtech — CHANGES_REQUESTED** [review]

File: src/components/elements/text/text.tsx (hunk 1 above)
Comment (wx-uf) [reply src/components/elements/text/text.tsx]: removed it

File: src/components/elements/text/text.tsx (hunk 2 above)
Comment (wx-uf) [reply src/components/elements/text/text.tsx]: follow the doc, use defalut list + img tag

File: src/__tests__/components/elements/text/text.spec.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+import { fireEvent, prettyDOM, render, screen, waitFor } from "@testing-library/react";
```
Comment (weili-govtech) [inline src/__tests__/components/elements/text/text.spec.tsx:1]:
```suggestion
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
```

File: src/components/elements/text/text.tsx [hunk 4]
```diff
@@ -71,9 +72,9 @@ export const Text = (props: IGenericElementProps<ITextSchema>) => {
 				);
 			});
 		} else if (typeof children === "object") {
-			return Object.entries(children).map(([id, childSchema], index) => (
-				<Text key={index} id={id} schema={childSchema} />
-			));
+			return Object.entries(children).map(([id, childSchema], index) => {
+				return <Text key={index} id={id} schema={childSchema} />;
+			});
```
Comment (weili-govtech) [inline src/components/elements/text/text.tsx:77]: can revert this change

File: src/stories/4-elements/text/text.stories.tsx [hunk 5]
```diff
@@ -279,7 +279,14 @@ ParagraphText.args = {
...
 	uiType: "text-body",
-	children: "<p>This is a paragraph in a HTML string</p>",
+	children: "<p>his is a paragraph in a HTML string</p>",
+};
+
+export const ImageTextString = Template("image-text-string").bind({});
+ImageTextString.args = {
+	uiType: "text-body",
+	children:
+		"<img src='https://assets.life.gov.sg/lifesg/logo-lifesg.svg'> <br/> <p>The image above is added via img tag and this text is rendered in a p tag</p>",
```
Comment (weili-govtech) [inline src/stories/4-elements/text/text.stories.tsx:289]: actually with `p`, the `br` is not required anymore
```suggestion
		"<img src='https://assets.life.gov.sg/lifesg/logo-lifesg.svg'><p>The image above is added via img tag and this text is rendered in a p tag</p>",
```

**Review by weili-govtech — CHANGES_REQUESTED** [review]

File: src/__tests__/components/elements/text/text.spec.tsx (hunk 3 above)
Comment (wx-uf) [reply src/__tests__/components/elements/text/text.spec.tsx:1]: removed import not in use

File: src/stories/4-elements/text/text.stories.tsx (hunk 5 above)
Comment (wx-uf) [reply src/stories/4-elements/text/text.stories.tsx:289]: removed

File: src/components/elements/text/text.tsx (hunk 4 above)
Comment (wx-uf) [reply src/components/elements/text/text.tsx:77]: reverted

**Review by weili-govtech — APPROVED** [review]
